### ISSUE TITLE
optimize GPU memory

### ISSUE THREAD

**Issue by SHYuanBest** [camenduru/consisid-tost#2]
thanks for your great support, you can add the following codes to optimize GPU memory:
```
pipe.enable_sequential_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()
```

**Comment by camenduru** [comment]
maybe this is better

```py
# pipe.to(device)
pipe.enable_model_cpu_offload()
pipe.enable_sequential_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()
```

**Comment by camenduru** [comment]
with 3090 + pipe.enable_sequential_cpu_offload() 

![Screenshot 2024-11-28 061221](https://github.com/user-attachments/assets/fec9fffe-54d2-4f26-92fb-08680aee84c8)

**Comment by SHYuanBest** [comment]
what is the input image, will it significantly reduce the quality?

**Comment by camenduru** [comment]
input image: ![wy4gq7](https://github.com/user-attachments/assets/35a7aeec-c7d8-447f-bbef-6e2341ee6d0b)

Enabling pipe.enable_sequential_cpu_offload() causes the inference time to increase from 350 to 720 seconds, but reduces the VRAM usage to 13GB 🔥

**Comment by camenduru** [comment]
``` py
input = { 
        "input": {
            "input_image": "https://files.catbox.moe/wy4gq7.jpg",
            "prompt": "A woman adorned with a delicate flower crown, is standing amidst a field of gently swaying wildflowers. Her eyes sparkle with a serene gaze, and a faint smile graces her lips, suggesting a moment of peaceful contentment. The shot is framed from the waist up, highlighting the gentle breeze lightly tousling her hair. The background reveals an expansive meadow under a bright blue sky, capturing the tranquility of a sunny afternoon.",
            "num_inference_steps": 50,
            "guidance_scale": 7.0,
            "seed": 42
        }
}
```

with

```py
pipe.enable_sequential_cpu_offload()
pipe.vae.enable_slicing()
pipe.vae.enable_tiling()
```

https://github.com/user-attachments/assets/2f205e0f-0c96-4258-ade2-2848460804ce

without

https://github.com/user-attachments/assets/da455617-d6ff-4a20-aa44-e01a2a225c95

**Comment by SHYuanBest** [comment]
It seems that the accuracy and time are exchanged for GPU memory.

**Comment by camenduru** [comment]
I realized that I forgot to add this part to the code (with pipe.enable_sequential_cpu_offload()) 😕

```py
tensor = align_crop_face_image.cpu().detach()
tensor = tensor.squeeze()
tensor = tensor.permute(1, 2, 0)
tensor = tensor.numpy() * 255
tensor = tensor.astype(np.uint8)
image  = ImageOps.exif_transpose(Image.fromarray(tensor))
```

https://github.com/user-attachments/assets/31cf7c44-af60-4b42-a2a1-a7f5bbe36ebb

It's working now 🥳

**Comment by SHYuanBest** [comment]
that's great!
